### PR TITLE
Update actions/cache to v2

### LINKED ISSUE #5362: Switch to actions/cache@v2
Cache actions got a new major version:

https://github.com/actions/cache

We should switch our CI to using it.

### PR DESCRIPTION
**Pull request by JohnTitor**
It has some improvements.

Fixes #5362

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -65,19 +65,19 @@ jobs:
...
       with:
         path: ~/.cargo/git
         key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.lock') }}
 
     - name: Cache cargo target dir
-      uses: actions/cache@v1
+      uses: actions/cache@v2
       with:
         path: target
         key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
```
Comment (matklad) [inline .github/workflows/ci.yaml:83]: v2 allows specifying the list of paths, so we can merge all three entries into one I think.

Comment (JohnTitor) [reply .github/workflows/ci.yaml:83]: Yep, we could do that way! But the `target` dir size will be big sometimes so I'd recommend to separate its key and `.cargo` dir stuff, how about?

Comment (matklad) [reply .github/workflows/ci.yaml:83]: Could elaborate a bit why that would be beneficial? I think I don’t understand something: we’ll still have to download/upload both, no?

We have https://github.com/rust-analyzer/rust-analyzer/blob/0f8af1b8246a31b1df60d3c60680954aefc9ab27/xtask/src/lib.rs#L123 to keep the size of target smaller, but that doesn’t sync with the list of crates, so it is bitrotted  atm :(

Comment (matklad) [reply .github/workflows/ci.yaml:83]: Note that https://github.com/rust-analyzer/rust-analyzer/pull/5361 adds a couple more caches. I think #5361 doesn't need to cache `./target`, and that actually is a good reason to split .cargo and target into different keys!

Comment (JohnTitor) [reply .github/workflows/ci.yaml:83]: > Could elaborate a bit why that would be beneficial?

In the past, we often encountered cache size limit and unpacking issues so it was a workaround to reduce per-cache size. And, in v2, sometimes it takes less time to upload a large cache in two parts than it does to upload it in one go AFAIK.
Anyway, given #5361, I agree to separate them, I'll update it shortly.

**Pull request merged**